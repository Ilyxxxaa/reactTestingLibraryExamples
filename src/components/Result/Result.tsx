import { useState, useEffect } from 'react';
import styled from 'styled-components';
import { getRandomColor } from './helpers';

type IProps = { name: string; surname: string };

const Result: React.FC<IProps> = ({ name, surname }) => {
  const [nameColor, setNameColor] = useState('black');
  useEffect(() => {
    const color = getRandomColor()
    setNameColor(color)
  }, [name]);

  const [surnameColor, setSurnameColor] = useState('black');
  useEffect(() => {
    const color = getRandomColor()
    setSurnameColor(color)
  }, [surname]);

  return (
    <Container>
      <Field color={nameColor}>Name - {name}</Field>
      <Field color={surnameColor}>Surname - {surname}</Field>
    </Container>
  )
}

const Container = styled.div`
  flex: 1;
  margin: 0 10px;
  justify-content: center;
  display: flex;
  flex-direction: column;
`;
const Field = styled.div`
  border: 3px solid ${({color}) => color};
  margin: 10px;
  padding: 10px;
  transition: border 1s;
`

export default Result;