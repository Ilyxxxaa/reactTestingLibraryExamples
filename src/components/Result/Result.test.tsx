import Result from './Result';
import { render, screen } from '@testing-library/react';
import { getRandomColor } from './helpers';

jest.mock('./helpers', () => ({
  getRandomColor: jest.fn(),
}))

const name = 'Johnny';
const surname = 'Silverhand';

const setup = () => {
  const utils = render(<Result name={name} surname={surname} />)

  return utils;
}

describe('Input component', () => {
  it('render name and surname', () => {
    setup();
    const nameText = screen.getByText(`Name - ${name}`);
    const surnameText = screen.getByText(`Surname - ${surname}`);

    expect(nameText).toBeInTheDocument();
    expect(surnameText).toBeInTheDocument();
  });

  it('should change border color', () => {
    (getRandomColor as jest.Mock).mockReturnValue('black');
    const { container, rerender } = setup();
    expect(container).toMatchSnapshot();

    (getRandomColor as jest.Mock).mockReturnValue('red');
    rerender(<Result name='new name' surname={surname} />);
    expect(container).toMatchSnapshot();

    (getRandomColor as jest.Mock).mockReturnValue('blue');
    rerender(<Result name='new name' surname='new surname' />);
    expect(container).toMatchSnapshot();
  });
});