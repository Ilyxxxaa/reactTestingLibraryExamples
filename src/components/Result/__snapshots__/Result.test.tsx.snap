// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should change border color 1`] = `
<div>
  <div
    class="sc-bcXHqe jzNJFC"
  >
    <div
      class="sc-gswNZR gGndHZ"
      color="black"
    >
      Name - 
      Johnny
    </div>
    <div
      class="sc-gswNZR gGndHZ"
      color="black"
    >
      Surname - 
      Silverhand
    </div>
  </div>
</div>
`;

exports[`Input component should change border color 2`] = `
<div>
  <div
    class="sc-bcXHqe jzNJFC"
  >
    <div
      class="sc-gswNZR csldCJ"
      color="red"
    >
      Name - 
      new name
    </div>
    <div
      class="sc-gswNZR gGndHZ"
      color="black"
    >
      Surname - 
      Silverhand
    </div>
  </div>
</div>
`;

exports[`Input component should change border color 3`] = `
<div>
  <div
    class="sc-bcXHqe jzNJFC"
  >
    <div
      class="sc-gswNZR csldCJ"
      color="red"
    >
      Name - 
      new name
    </div>
    <div
      class="sc-gswNZR dNPVBI"
      color="blue"
    >
      Surname - 
      new surname
    </div>
  </div>
</div>
`;
